lib/v2/modrinth/versions.js

```diff
@@ -0,0 +1,84 @@
+const { parseDate } = require('@/utils/parse-date');
+const { art } = require('@/utils/render');
+const path = require('path');
+const { value: config } = require('@/config');
+const got = require('@/utils/got').extend({
+    headers: {
+        // https://docs.modrinth.com/#section/User-Agents
+        'user-agent': config.trueUA,
+    },
+});
+const markdownIt = require('markdown-it')({
+    html: true,
+});
+
+module.exports = async (
+    /**
+     * @type {import('koa').Context}
+     */
+    ctx
+) => {
+    const {
+        /**
+         * @type string
+         */
+        id,
+    } = ctx.params;
+    let {
+        /**
+         * @type {string | string[] | undefined}
+         */
+        loaders,
+        /**
+         * @type {string | string[] | undefined}
+         */
+        game_versions,
+    } = ctx.query;
+    const { featured } = ctx.query;
+
+    if (typeof loaders === 'string') {
+        loaders = [loaders];
+    }
+    if (typeof game_versions === 'string') {
+        game_versions = [game_versions];
+    }
+
+    try {
+        /**
+         * @type {import('./api').Project}
+         */
+        const project = await got(`https://api.modrinth.com/v2/project/${id}`).json();
+        /**
+         * @type {import('./api').Version[]}
+         */
+        const versions = await got(
+            `https://api.modrinth.com/v2/project/${id}/version?${loaders ? `loaders=${JSON.stringify(loaders)}` : ''}${game_versions ? `&game_versions=${JSON.stringify(game_versions)}` : ''}&${featured === undefined ? '' : `&featured=${featured}`}`
+        ).json();
+        /**
+         * @type {import('./api').Author[]}
+         */
+        const authors = await got(`https://api.modrinth.com/v2/users?ids=${JSON.stringify(versions.map((it) => it.author_id))}`).json();
```

Comment:
```suggestion
        /** @type {import('./api').Project} */
        const project = await got(`https://api.modrinth.com/v2/project/${id}`).json();
        /** @type {import('./api').Version[]} */
        const versions = await got(`https://api.modrinth.com/v2/project/${id}/version`, {
            searchParams: {
                loaders: JSON.stringify(loaders),
                game_versions: JSON.stringify(game_versions),
                featured,
            },
        }).json();
        /** @type {import('./api').Author[]} */
        const authors = await got(`https://api.modrinth.com/v2/users`, {
            searchParams: {
                ids: JSON.stringify(versions.map((it) => it.author_id)),
            },
        }).json();
```
You can use the [`searchParams option`](https://github.com/sindresorhus/got/tree/v11.8.6?tab=readme-ov-file#searchparams) so that it will handle [URI encoding properly](https://github.com/DIYgod/RSSHub/actions/runs/7700719619/job/20985147585#step:12:17).